tailscale/client.go

```diff
@@ -333,6 +332,62 @@ func (c *Client) DNSNameservers(ctx context.Context) ([]string, error) {
 	return resp["dns"], nil
 }
 
+// SplitDnsRequest is a map from domain names to a list of nameservers.
+type SplitDnsRequest map[string][]string
+
+// SplitDnsResponse is a map from domain names to a list of nameservers.
+type SplitDnsResponse SplitDnsRequest
+
+// UpdateDNSSplitDNS updates the split DNS settings for a tailnet using the
+// passed in SplitDnsRequest object. This is a PATCH operation that performs
+// partial updates of the underlying data structure.
+//
+// Mapping a domain to a nil slice in the request will unset the nameservers
+// associated with that domain.
+func (c *Client) UpdateDNSSplitDNS(ctx context.Context, request SplitDnsRequest) error {
```

Comment:
Open to different names here, was following existing convention within this file.

Reply:
Sounds good to me! Definitely agree on the readability front.